### ISSUE TITLE
Retrieve input history and Display on the sidebar

### ISSUE THREAD

**Issue by ChTiSh** [Linkta-org/core#148]
## Related to User Story 
#120 

## Corresponding backend ticket
#152 #178 

## Highly Related to 
#149 #121 #125 #178 

# Subtasks
#159 #160 #182 #204 

## Description ##
I need to retrieve the user input history from the backend API and display it in the frontend. Each input history item should be rendered as a clickable link and have a hamburger menu for additional options.

### Acceptance criteria
- Add a new API service or method to retrieve the input history from the backend.
- Render the input history as links within the component.
- Implement the hamburger menu functionality for each input history item.

**Comment by kinskiwu** [comment]
#159 in review. #160 & #178 will be worked on next.

**Comment by kinskiwu** [comment]
#160 in review. Work on #182 next.

**Comment by kinskiwu** [comment]
#159 #160 merged. Only outstanding subtasks are incorporating react query & zustand into this feature (ticket yet to be made), styling(#182) and updating the UI and fetch logic (ticket yet to be made) after #178 is implemented.

**Comment by kinskiwu** [comment]
#182 in review. New ticket #204 created for RQ & Zustand implementation. Update UI and fetch logic for input history will be implemented in #178.

**Comment by kinskiwu** [comment]
Putting this issue to review given that #204 is in review.
